app/main/browser.js

```diff
@@ -38,100 +38,128 @@ import * as instrument from '../services/instrument';
 import * as spawn from './spawn';
 import * as BW from './browser-window';
 import UserAgentClient from '../shared/user-agent-client';
+import { argParser, parseArgs } from '../shared/environment';
+
+const SQUIRREL_EVENTS = [
+  'squirrel-install',
+  'squirrel-updated',
+  'squirrel-uninstall',
+  'squirrel-obsolete',
+];
+argParser.boolean(SQUIRREL_EVENTS);
+
+// For now we don't actually do anything on squirrel events, just let the app
+// exit.
+function handledSquirrelEvent() {
+  const options = parseArgs();
+
+  for (const event of SQUIRREL_EVENTS) {
+    if (options[event]) {
+      return true;
+    }
+  }
+
+  return false;
+}
 
 const app = electron.app; // control application life.
-const ipc = electron.ipcMain;
-const globalShortcut = electron.globalShortcut;
-const userAgentClient = new UserAgentClient();
 
-const appStartupTime = Date.now();
-instrument.event('app', 'STARTUP');
+if (handledSquirrelEvent()) {
+  app.quit();
+} else {
```

Comment:
AFAICT everything below here is just whitespace changes, correct me if that's wrong